internal/datastore/mysql/migrations/driver.go

```diff
@@ -0,0 +1,124 @@
+package migrations
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
+
+	sq "github.com/Masterminds/squirrel"
+	sqlDriver "github.com/go-sql-driver/mysql"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	errUnableToInstantiate       = "unable to instantiate MysqlDriver: %w"
+	mysqlMissingTableErrorNumber = 1146
+
+	migrationVersionColumnPrefix = "_meta_version_"
+)
+
+var sb = sq.StatementBuilder.PlaceholderFormat(sq.Question)
+
+type MysqlDriver struct {
+	db *sql.DB
+	*tables
+}
+
+// NewMysqlDriverFromDSN creates a new migration driver with a connection pool to the database DSN specified.
+//
+// URI: [scheme://][user[:[password]]@]host[:port][/schema][?attribute1=value1&attribute2=value2...
+// See // https://dev.mysql.com/doc/refman/8.0/en/connecting-using-uri-or-key-value-pairs.html
+func NewMysqlDriverFromDSN(url string, tablePrefix string) (*MysqlDriver, error) {
+	dbConfig, err := sqlDriver.ParseDSN(url)
+	if err != nil {
+		return nil, fmt.Errorf(errUnableToInstantiate, err)
+	}
+
+	db, err := sql.Open("mysql", dbConfig.FormatDSN())
+	if err != nil {
+		return nil, fmt.Errorf(errUnableToInstantiate, err)
+	}
+	err = sqlDriver.SetLogger(&log.Logger)
+	if err != nil {
+		return nil, fmt.Errorf("unable to set logging to mysql driver: %w", err)
+	}
+	return NewMysqlDriverFromDB(db, tablePrefix), nil
+}
+
+// NewMysqlDriverFromDB creates a new migration driver with a connection pool specified upfront.
+func NewMysqlDriverFromDB(db *sql.DB, tablePrefix string) *MysqlDriver {
```

Comment:
is the `tablePrefix` for the pool, or...?

Reply:
`tablePrefix` is necessary to have the migrate driver generate DML statements that take into consideration the table-prefix